src/Aspire.Dashboard/Components/Controls/SelectResourceOptions.razor

```diff
@@ -14,10 +14,10 @@
                     ShowIndeterminate="false"
                     ThreeStateOrderUncheckToIntermediate="true"
                     @bind-CheckState:get="@GetCheckState(Values)"
-                    @bind-CheckState:set="@OnAllValuesCheckedChangedInternalAsync"
-    />
+                    @bind-CheckState:set="@OnAllValuesCheckedChangedInternalAsync" />
 
-    @foreach (var (key, isChecked) in Values.OrderBy(pair => pair.Key.ToString(), StringComparer.OrdinalIgnoreCase))
+    @* OrderBy doesn't use thread safe APIs on ConcurrentDictionary. Call ToArray first. *@
+    @foreach (var (key, isChecked) in Values.ToArray().OrderBy(pair => pair.Key.ToString(), StringComparer.OrdinalIgnoreCase))
```

Comment:
@stephentoub this seem right to you?

Reply:
LINQ operators that need to buffer will try to use a collection's `ICollection<T>`, separately calling Count and CopyTo. This is true even on netfx, eg 
https://github.com/microsoft/referencesource/blob/main/System.Core/System/Linq/Enumerable.cs#L2674-L2679

https://github.com/dotnet/runtime/issues/14502

Reply:
The issue says `ToArray` has the same problem, so calling `ToArray` before `OrderBy` doesn't fix this. (I thought it would use the enumerator internally so not have the problem)

@stephentoub What do you recommend to make this safe? Have a method that uses a foreach to copy values to a list and then do LINQ on that list? Or maybe switch to using ImmutableDictionary instead of ConcurrentDictionary?

Reply:
ConcurrentDictionary has its own ToArray method that is safe and which I'm unintentionally using.

https://learn.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.toarray?view=net-9.0

This PR will fix the problem them.